src/compiler/checker.ts

```diff
@@ -1903,7 +1903,13 @@ namespace ts {
         }
 
         function getTargetOfExportAssignment(node: ExportAssignment, dontResolveAlias: boolean): Symbol {
-            return resolveEntityName(<EntityNameExpression>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace, /*ignoreErrors*/ false, dontResolveAlias);
+            const aliasLike = resolveEntityName(<EntityNameExpression>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace, /*ignoreErrors*/ true, dontResolveAlias);
+            if (aliasLike) {
+                return aliasLike;
+            }
+            const links = getNodeLinks(node.expression);
+            checkExpression(node.expression);
+            return links.resolvedSymbol;

```

Comment:
Should the return type be `Symbol | undefined`? Presumably only a few expression kinds set this property.
Also, can't this just `return getNodeLinks(node.expression).resolvedSymbol`?